Add tabs to the categories view

adrianz-kainos opened the pull request:
Resolves #30

adrianz-kainos committed:
Add tabs to the categories view

xjedam commented on views/pages/categoryRadar.ejs:
```diff
@@ -40,38 +40,73 @@
+                    if( technology.status != heading) {
+                        heading = technology.status;
+
+                    if(i != 0) { %>
```
Very convoluted logic, is it possible to simplify/comment on some of the code here?

adrianz-kainos committed:
Merge branch 'master' into feature/add-tabs-to-categories

* master: (34 commits)
  Use tab('show') instead of click() to switch tabs Change the colour of the highlighted technology
  Fix popup message
  Remove unnecessary SQL clause
  Add "Tags" option to "Projects" dropdown
  Fix typo
  Add missing error handler in projects DAO
  Add unit tests for tags API handler Remove unnecessary (app) argument in tags API handler Fix typo in passport logger Add sticky options to the tags table
  fix empty category screen, fix tests
  fix connection pooling for pg version 6 on node 6
  fix pg version to the newest one
  change node and npm version to support newest sub-dependency versions
  Use a bootstrap modal to edit tags
  Fix avatar not loading for adfs users on page with comments
  Use a bootstrap modal to add new tags Add Google Analytics tracking
  Use a single screen to reassign tags. Edit tags on a separate page.
  Refactor code Remove redundant script tag Use apiutils Remove duplicate element id from projectRadar
  Refactor attachToProject() in tags DAO, Use dbhelper in Tag.getAll
  Add a shared validator for tags
  Switch to .js file instead of .ejs
  Add a separate template for updating comments Use strict comparison operators
  ...

adrianz-kainos committed:
Put the radar list in a partial
Simplify categoryRadar.ejs
Use underscore to group technologies by status

adrianz-kainos commented on package.json:
```diff
@@ -22,33 +22,34 @@
-    "express-validator" : "latest",
-    "connect-flash" : "0.1.1"
+    "sinon": "latest",
+    "underscore": "^1.8.3"
```
FYI: I only added the underscore dependency. The remaining changes were made by npm auto-sorting.

xjedam replied:
👍

adrianz-kainos committed:
Use the reworked tech list in projectRadar.ejs
Assign numbers to the same technologies using SQL
Highlight all technologies related to the clicked radar circle

adrianz-kainos commented on handlers/web/categoriesWebHandler.js:
```diff
@@ -26,8 +27,24 @@ CategoriesWebHandler.technologiesForCategory = function (req, res) {
+        res.render('pages/categoryRadar', {
+            category: category,
+            technologies: values, // used by radar.js
+            technologiesInGroups: technologiesInGroups,
```
`technologiesInGroups` and  `technologies` contain the same data (structured differently). I couldn't get rid of `technologies` without reworking radar.js. Changing radar.js to use `technologiesInGroups` might take a whole workday. Is it worth it?

xjedam replied:
It looks great now :)

adrianz-kainos commented on views/pages/categoryRadar.ejs:
```diff
@@ -5,19 +5,16 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
-    <meta name="description" content="">
```
`meta name="description"` is in `head.ejs`
